### ISSUE TITLE
Support SSH authentication for git artifacts

### ISSUE THREAD

**Issue by alexmt** [argoproj/argo#793]
**Is this a BUG REPORT or FEATURE REQUEST?**:

FEATURE REQUEST

**What happened**:

HTTPS and password authentication might be disabled. Argo should support authentication using ssh key.

**What you expected to happen**:

It should be possible to define git artifact using 'git@' url and ssh key secret:

```
        git:
          repo: "git@github.com:argoproj/argo.git"
          revision: "master"
          sshKeySecret: 'argo-github-ssh'
```

**Comment by andreimc** [comment]
https://github.com/argoproj/argo/pull/919 this does all 3 basic auth private key and plain using go-git

**Comment by andreimc** [comment]
@alexmt can you check it out and merge if all good?

**Comment by andreimc** [comment]
@jessesuen this can be closed with #919 merge

**Comment by jessesuen** [comment]
Thanks @andreimc. 

Fixed in PR #919, commit 4744f45a9c110b11fa73070a52e4166406fa5da4
